### DISCUSSION TITLE
conditinal checking of pattern if group was matched

### DISCUSSION THREAD

**Discussion by swanandx** [rust-lang/regex#1162] [Q&A] [answered]
I needed to check for a specific pattern only if there was group matched previously, so for example, only check if `x` is present if `<me>` group is matched, otherwise `z` should be present.

syntax for this in Python would be something like:
```
(user|(?P<me>swan))(?(me)x|z)
```

important part here is:
```
?(group_name)pattern_to_check_if_group_was_matched | otherwise_this 
```

can we support this syntax / feature in `regex`?

ps: hope I'm able to convey it, couldn't think of better example / technical terms :sweat_smile:

**Comment by BurntSushi** [comment, marked as answer]
No, sorry. This looks like a variant of backreferences to me. The opening paragraph of this crate's docs says:

> This crate provides routines for searching strings for matches of a [regular expression](https://en.wikipedia.org/wiki/Regular_expression) (aka “regex”). The regex syntax supported by this crate is similar to other regex engines, but it lacks several features that are not known how to implement efficiently. This includes, but is not limited to, look-around and backreferences. In exchange, all regex searches in this crate have worst case O(m * n) time complexity, where m is proportional to the size of the regex and n is proportional to the size of the string being searched.
